mephisto/client/review/review_server.py

```diff
@@ -96,18 +96,32 @@ def consume_data():
             ready_for_next.wait()
         finished = True
 
-    def consume_all_data(page, limit=RESULTS_PER_PAGE_DEFAULT):
-        """ For use in "all" mode. Returns all data or a page of all data given a limit where pages are 1 indexed """
+    def consume_all_data(page, results_per_page=RESULTS_PER_PAGE_DEFAULT, filters=None):
+        """
+        For use in "all" mode.
+        Returns a filtered list of all data or a page of all data.
+        Params:
+            page: 1 indexed page number integer
+            results_per_page: maximum number of results per page
+            filters: keywords or sentences to filter data for. must be a list
+        """
         global all_data_list, datalist_update_time
         paginated = type(page) is int
         if paginated:
             assert page > 0, "Page number should be a positive 1 indexed integer."
             assert (
-                type(limit) is int and limit > 0
+                type(results_per_page) is int and results_per_page > 0
             ), "results_per_page should be a positive integer"
 
-        first_index = (page - 1) * limit if paginated else 0
-        data_point_list = []
+        first_index = (page - 1) * results_per_page if paginated else 0
+
+        filtered_data_list = all_data_list
+        if type(filters) is list:
+            filtered_data_list = [
+                item
+                for item in all_data_list
+                if all(word.lower() in str(item["data"]).lower() for word in filters)
```

Comment:
Note: if search becomes a bigger thing, you may eventually need to swap to creating a term index. Not really necessary right now, but perhaps to look out for as this gets larger.

Reply:
Yea I was definitely considering this but figured it might be overkill. I might ask how much data Ego4d plans on manipulating and upgrading this later.